drivers/firmware/scmi/core.c

```diff
@@ -149,11 +149,17 @@ static int scmi_send_message_post_kernel(struct scmi_protocol *proto,
 		goto out_release_mutex;
 	}
 
-	/* only one protocol instance can wait for a message reply at a time */
-	ret = k_sem_take(&proto->tx->sem, K_USEC(SCMI_CHAN_SEM_TIMEOUT_USEC));
-	if (ret < 0) {
-		LOG_ERR("failed to wait for msg reply");
-		goto out_release_mutex;
+	if (!scmi_transport_channel_is_polling(proto->transport, proto->tx)) {
```

Comment:
All transport channels should support polling mode because the `send_message` function is forced to poll in `PRE_KERNEL` so this function should always return `true`. As such, I suggest renaming to `scmi_transport_channel_is_polling_only`.